src/Service.Tests/Unittests/ConfigValidationUnitTests.cs

```diff
@@ -631,6 +633,323 @@ string linkingObject
+
+            // Exception is thrown since either source or target field does not exist as a valid backing column in their respective entity.
+            DataApiBuilderException ex = Assert.ThrowsException<DataApiBuilderException>(() =>
+                configValidator.ValidateRelationshipsInConfig(runtimeConfig, _metadataProviderFactory.Object));
+            Assert.AreEqual(expectedExceptionMessage, ex.Message);
+            Assert.AreEqual(HttpStatusCode.ServiceUnavailable, ex.StatusCode);
+            Assert.AreEqual(DataApiBuilderException.SubStatusCodes.ConfigValidationError, ex.SubStatusCode);
+        }
+
+        [DataRow(
+            new string[] { "sourceField" },
+            new string[] { "non null" },
+            new string[] { "targetField" },
+            null,
+            "SampleEntity2",
+            "Entity: SampleEntity1 has a relationship: rname1 with linking source fields that are not null, " +
+                "but linking target fields that are null.",
+            DisplayName = "Linking source fields are non null, but linking target fields are null.")]
+        [DataRow(
+            new string[] { "sourceField" },
+            null,
+            new string[] { "targetField" },
+            new string[] { "non null" },
+            "SampleEntity2",
+            "Entity: SampleEntity1 has a relationship: rname1 with linking target fields that are not null, " +
+                "but linking source fields that are null.",
+            DisplayName = "Linking target fields are non null, but linking source fields are null.")]
+        [DataRow(
+            new string[] { "sourceField" },
+            new string[] { "A", "B", "C" },
+            new string[] { "targetField" },
+            new string[] { "1", "2" },
+            "SampleEntity2",
+            "Entity: SampleEntity1 has 3 linking source fields defined, but 2 linking target fields defined.",
+            DisplayName = "Linking source fields and linking target fields are different length.")]
+        [DataTestMethod]
+        public void TestRelationshipWithoutLinkingSourceAndTargetFieldsMatching(
+            string[] sourceFields,
+            string[] linkingSourceFields,
+            string[] targetFields,
+            string[] linkingTargetFields,
+            string relationshipEntity,
+            string expectedExceptionMessage)
+        {
+            // Creating an EntityMap with two sample entity
+            Dictionary<string, Entity> entityMap = GetSampleEntityMap(
+                sourceEntity: "SampleEntity1",
+                targetEntity: "SampleEntity2",
+                sourceFields: sourceFields,
+                targetFields: targetFields,
+                linkingObject: "TEST_SOURCE_LINK",
+                linkingSourceFields: linkingSourceFields,
+                linkingTargetFields: linkingTargetFields
+            );
+
+            RuntimeConfig runtimeConfig = new(
```

Comment:
The code for creating a runtimeconfig seems to be duplicate across tests. Can this be moved to some common helper? Also, other existing tests which also create config in a similar manner can reuse that helper.

Reply:
There is a lot of code duplication in our tests, which is OK, but also should be refactored out for more readability. I will either refactor this out, or maybe include this in a larger issue that tracks readability for tests more generally, as this is one of many, many cases and as you point out includes other tests in this class too.

Reply:
Makes sense. Please open up an issue tracking this refactor.

Reply:
issue here https://github.com/Azure/data-api-builder/issues/2143

Reply:
#2143